src/drivers/CgreenDriver.cpp

```diff
@@ -0,0 +1,159 @@
+#include <cucumber-cpp/internal/drivers/CgreenDriver.hpp>
+
+#include <boost/bind.hpp>
+#include <boost/function.hpp>
+
+using namespace cgreen;
+
+namespace cucumber {
+namespace internal {
+
+
+bool CgreenStep::initialized = false;
+
+static const char* const blacklist[] = {
+        "Running",
+        "Completed",
+        "Failure",
+        "currentTest",
+        "\n",
+        "/"
+};
+
+class CukeCgreenInterceptor {
```

Comment:
Correct, you should assume only C++98, as we currently still are able to build with just that. You don't need `std::unique_ptr` for a singleton though, that just makes it slightly easier to implement.